integration-testing/build.gradle.kts

```diff
@@ -0,0 +1,113 @@
+import org.jetbrains.kotlin.gradle.utils.NativeCompilerDownloader
+
+buildscript {
+
+    /*
+     * These property group is used to build kotlinx-atomicfu against Kotlin compiler snapshot.
+     * How does it work:
+     * When build_snapshot_train is set to true, kotlin_version property is overridden with kotlin_snapshot_version,
+     * atomicfu_version is overwritten by TeamCity environment (AFU is built with snapshot and published to mavenLocal
+     * as previous step or the snapshot build).
+     * Additionally, mavenLocal and Sonatype snapshots are added to repository list and stress tests are disabled.
+     * DO NOT change the name of these properties without adapting kotlinx.train build chain.
+     */
+    val prop = project.properties["build_snapshot_train"]
+    val build_snapshot_train = prop != null && prop != ""
+    val kotlin_version = project.properties["kotlin_snapshot_version"]
+    if (build_snapshot_train) {
+        if (kotlin_version == null) {
+            throw IllegalArgumentException("'kotlin_snapshot_version' should be defined when building with snapshot compiler")
+        }
+    }
+
+    // Determine if any project dependency is using a snapshot version
+    var using_snapshot_version = build_snapshot_train
+    project.properties.forEach {
+        if (it.key.endsWith("_version") && (it.value as String).endsWith("-SNAPSHOT")) {
+            println("NOTE: USING SNAPSHOT VERSION: ${it.key}=${it.value}")
+            using_snapshot_version = true
+        }
+    }
+
+    if (using_snapshot_version) {
+        repositories {
+            mavenLocal()
+            maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev")
+        }
+    }
+}
+
+plugins {
+    kotlin("jvm") version "${project.properties["kotlin_version"]}" // todo: is there a more consise way to get kotlin_version from gradle.properties?
```

Comment:
Maybe [versions catalog](https://developer.android.com/build/migrate-to-catalogs)?

Reply:
When I configure version catalog like this, I get "The root project is not yet available for build."  👀

```
dependencyResolutionManagement {
    versionCatalogs {
        create("libs") {
            version("atomicfu", gradle.rootProject.properties["atomicfu_version"])
        }
   }
}
```